docker/models/images.py

```diff
@@ -169,14 +169,15 @@ def build(self, **kwargs):
         events = list(json_stream(resp))
         if not events:
             return BuildError('Unknown')
-        event = events[-1]
-        if 'stream' in event:
-            match = re.search(r'(Successfully built |sha256:)([0-9a-f]+)',
-                              event.get('stream', ''))
-            if match:
-                image_id = match.group(2)
-                return self.get(image_id)
+        for event in events:
```

Comment:
If we're going through all events anyway, the conversion to list we do above is wasteful. We should iterate on `json_stream(resp)` directly.